tests/functional/test_wheel.py

```diff
@@ -210,3 +217,31 @@ def test_pip_wheel_with_user_set_in_config(script, data, common_wheels):
         '--no-index', '-f', common_wheels
     )
     assert "Successfully built withpyproject" in result.stdout, result.stdout
+
+
+def test_pep517_wheels_are_not_confused_with_other_files(script, tmpdir, data):
+    """Check correct wheels are copied. (#6196)
+    """
+    pkg_to_wheel = data.src / 'withpyproject'
+    add_files_to_dist_directory(pkg_to_wheel)
+
+    result = script.pip('wheel', pkg_to_wheel, '-w', script.scratch_path)
+    assert "Installing build dependencies" in result.stdout, result.stdout
+
+    wheel_file_name = 'withpyproject-0.0.1-py%s-none-any.whl' % pyversion[0]
+    wheel_file_path = "scratch/" + wheel_file_name
+    assert wheel_file_path in result.files_created, result.stdout
+
+
+def test_legacy_wheels_are_not_confused_with_other_files(script, tmpdir, data):
+    """Check correct wheels are copied. (#6196)
+    """
+    pkg_to_wheel = data.src / 'simplewheel-1.0'
+    add_files_to_dist_directory(pkg_to_wheel)
+
+    result = script.pip('wheel', pkg_to_wheel, '-w', script.scratch_path)
+    assert "Installing build dependencies" not in result.stdout, result.stdout
```

Comment:
I would assert something positive here or else make `"Installing build dependencies"` a variable that's used in both this test and the previous. Otherwise, if the message ever changes, the test can continue to pass but for the wrong reason.

Reply:
That's the same pattern as existing test; we can fix this in a follow up.